File: Selenium/selenium-todo-component-tester/src/test/java/no/ahlstroem/pages/AddPageObject.java
package no.ahlstroem.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class AddPageObject {

    private WebDriver webDriver;

    @FindBy(id="description-field")
    private WebElement descriptionField;

    @FindBy(id="cancel-button")
    private WebElement cancelButton;

    @FindBy(id="save-button")
    private WebElement saveButton;

    public AddPageObject(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(webDriver, this);
    }

    public AddPageObject fillDescriptionField(String description) {
        descriptionField.sendKeys(description);
        return this;
    }

    public HomePageObject clickCancelButton() {
        cancelButton.click();
        return new HomePageObject(webDriver);
    }

    public HomePageObject clickSaveButton() {
        saveButton.click();
        return new HomePageObject(webDriver);
    }

 }

File: todo-app/src/test/pageobjects/AddPageObject.ts
import { fireEvent, screen } from "@testing-library/react";
import HomePageObject from "./HomePageObject";

class AddPageObject {

private readonly descriptionField: HTMLElement;
private readonly cancelButton: HTMLElement;
private readonly saveButton: HTMLElement;

  constructor() {
    this.descriptionField = screen.getByRole('textbox', {name: 'Description'});
    this.cancelButton = screen.getByRole('button', {name: 'Cancel'});
    this.saveButton = screen.getByRole('button', {name: 'Save'});
    this.verifyExistenceOfElements();
  }

  public fillDescriptionField(description: string) {
    fireEvent.change(this.descriptionField, {target: {value: description}});
    return this;
  }

  public clickCancelButton() {
    fireEvent.click(this.cancelButton);
    return new HomePageObject();
  }

  public clickSaveButton() {
    fireEvent.click(this.saveButton);
    return new HomePageObject();
  }

  private verifyExistenceOfElements() {
    if (!this.descriptionField) {
      fail('The description field has not been initialized');
    }
    if (!this.cancelButton) {
      fail('The cancel button has not been initialized');
    }
    if (!this.saveButton) {
      fail('The save button has not been initialized');
    }
  }
}

export default AddPageObject;